crates/red_knot_python_semantic/src/types/call.rs

```diff
@@ -89,16 +95,18 @@ impl<'db> CallOutcome<'db> {
                 })
                 .map(UnionBuilder::build),
             Self::PossiblyUnboundDunderCall { call_outcome, .. } => call_outcome.return_ty(db),
+            Self::StaticAssertionError { .. } => Some(Type::none(db)),
         }
     }
 
     /// Get the return type of the call, emitting default diagnostics if needed.
     pub(super) fn unwrap_with_diagnostic(
         &self,
+        db: &'db dyn Db,
```

Comment:
The db is already available as `context.db()`, we shouldn't need it as a separate argument.